### PR TITLE
added MVP integration of front and back

### PR DESCRIPTION
**Pull request by mineralsfree**

### CODE REVIEW

File: src/components/Map/Map.jsx [hunk 1]
```diff
@@ -7,14 +7,17 @@ import "leaflet-defaulticon-compatibility";
...
 import {ItemMarker} from "@/components/Map/ItemMarker";
 import L from "leaflet";
+import {SelectionOverlay} from "@/components/Map/SelectionOverlay";
+import {useChatMutation} from "@/store/services/chatApi";
 
 export const Map = (props) => {
   const {position: initialPosition} = props;
   const icon = L.icon({iconUrl: "/images/marker-icon.png"});
-
+  const [chat, { isLoading, isSuccess, data, error }] = useChatMutation();
```
Comment (sebazai) [inline src/components/Map/Map.jsx:16]: We should use /api/places endpoint here to get an answer like the constant data2 used.

File: src/app/constants.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const AIModel = "openhermes";
```
Comment (sebazai) [inline src/app/constants.js:1]: Update readme to pull openhermes

File: src/app/map/page.jsx [hunk 3]
```diff
@@ -1,13 +1,9 @@
 "use client";
 
-import dynamic from "next/dynamic";
 import { useEffect, useState } from "react";
+import { Map } from "@/components/Map/Map";
 
 // Remove SSR for MapComponent, Leaflet uses window requiring client-side rendering
-const MapComponent = dynamic(() => import("../../components/Map/Map"), {
```
Comment (sebazai) [inline src/app/map/page.jsx:7]: Why is this removed?

File: src/components/Map/Map.jsx [hunk 4]
```diff
@@ -7,14 +7,17 @@ import "leaflet-defaulticon-compatibility";
...
+  console.log(data);
   const [position, setPosition] = useState(initialPosition); // State to store user's position
   const [loading, setLoading] = useState(true);  // Loading state for geolocation
-
+  console.log(isLoading, isSuccess, error, data);
   useEffect(() => {
```
Comment (sebazai) [inline src/components/Map/Map.jsx:21]: This geolocation fetching is redundant, as parent component fetches it and passes it to map

**Comment by sebazai** [comment]
I would remove all logic from components and move them more into pages.  i.e. map/page.jsx
I also do the /api/places call in the page.jsx to fetch the data, which could be added to redux if seen necessary or in a state.

File: src/app/map/page.jsx (hunk 3 above)
Comment (mineralsfree) [reply src/app/map/page.jsx:7]: It was causing runtime errors and we don't need SSR for this app at all IMO

File: src/components/Map/Map.jsx (hunk 1 above)
Comment (mineralsfree) [reply src/components/Map/Map.jsx:16]: I've forgotten to remove `data2`, sorry

**Comment by mineralsfree** [comment]
> I would remove all backend related logic from components and move them more into pages. i.e. map/page.jsx I also do the /api/places call in the page.jsx to fetch the data, which could be added to redux if seen necessary or in a state.

Can be done as refactoring tasks later on :+1:

File: src/app/map/page.jsx (hunk 3 above)
Comment (sebazai) [reply src/app/map/page.jsx:7]: This actually removes the pre-rendering of the HTML page for NextJS that caused problems in leaflet, as it requires the "window" object from the client. What runtime error did you get?

**Review by sebazai — APPROVED** [review]
